src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1808,21 +1847,37 @@ public static void shutdownAndWait(long timeout, TimeUnit units) throws Interrup
      * @param indexQueryPlan index query plan used in the read command
      * @param level consistency level of read command
      */
-    public static <E extends Endpoints<E>> E filterForQuery(E liveEndpoints, Keyspace keyspace, Index.QueryPlan indexQueryPlan, ConsistencyLevel level)
+    public static <E extends Endpoints<E>> E filterForQuery(E liveEndpoints,
+                                                            Keyspace keyspace,
+                                                            Index.QueryPlan indexQueryPlan,
+                                                            ConsistencyLevel level,
+                                                            boolean allowFiltering)
     {
-        E queryableEndpoints = liveEndpoints.filter(replica -> {
+        int initial = liveEndpoints.size();
+        Set<InetAddressAndPort> badNodes = MessagingService.instance().endpointsWithVersionBelow(MessagingService.VERSION_DS_11);
+        if (MessagingService.current_version < MessagingService.VERSION_DS_11)
+            badNodes.add(FBUtilities.getBroadcastAddressAndPort());
+
+        assert liveEndpoints.endpoints().containsAll(badNodes);
 
+        boolean considerAllowFiltering = badNodes.isEmpty();
+
+        E queryableEndpoints = liveEndpoints.filter(replica -> {
             for (Index index : indexQueryPlan.getIndexes())
             {
                 Index.Status status = getIndexStatus(replica.endpoint(), keyspace.getName(), index.getIndexMetadata().name);
+
+                // if the status of the index is building and there is allow filtering - that is ok too
+                if (considerAllowFiltering && status == Index.Status.INITIALIZED && allowFiltering)
```

Comment:
We may still want to check queryability because of CNDB. Though during INITIALIZED we should in theory always have the index not queryable there